crates/evm/traces/src/identifier/local.rs

```diff
@@ -32,17 +32,28 @@ impl<'a> LocalTraceIdentifier<'a> {
         self.known_contracts
     }
 
-    /// Tries to the bytecode most similar to the given one.
-    pub fn identify_code(&self, code: &[u8]) -> Option<(&'a ArtifactId, &'a JsonAbi)> {
-        let len = code.len();
+    /// Identifies the artifact based on score computed for both creation and deployed bytecodes.
+    pub fn identify_code(
+        &self,
+        runtime_code: &[u8],
+        creation_code: &[u8],
+    ) -> Option<(&'a ArtifactId, &'a JsonAbi)> {
+        let len = runtime_code.len();
 
         let mut min_score = f64::MAX;
         let mut min_score_id = None;
 
         let mut check = |id| {
             let contract = self.known_contracts.get(id)?;
             if let Some(deployed_bytecode) = contract.deployed_bytecode() {
-                let score = bytecode_diff_score(deployed_bytecode, code);
+                let mut score = bytecode_diff_score(deployed_bytecode, runtime_code);
+
+                // Compute and use creation code score.
+                if let Some(bytecode) = contract.bytecode() {
+                    score += bytecode_diff_score(bytecode, creation_code);
+                    score /= 2.0;
+                }
```

Comment:
do we actually need to check both here? are there cases when comparing just creation code is not enough?

Reply:
this looks like an edge case we've discussed earlier. constructor args are of length 192 while bytecode length is 1320. on >10% diff we consider bytecodes completely different.

what we could do is to only fallback to deployed bytecode comparison if we haven't found a match through comparing creation codes and/or target for exact match when comparing creation codes (i.e if we have an artifact with creation code that exactly matches the beginning of some creation bytecode in traces, then this is a 100% match)

Reply:
nice! yep, this is waht we want. could we refactor this a bit and still have a single `check` closure which would just accept a `is_creation: bool`? looks like we

Reply:
updated in https://github.com/foundry-rs/foundry/pull/9050/commits/781b892c1037d36c98d7bb8bff6eaf1b5d8a250c I also added a test for scenario mentioned above, ptal (unrelated tests failing due to avax rpc url) Thanks!